### ISSUE TITLE
Sage 10 - slow compiling time and no style injection

### ISSUE THREAD

**Issue by cfabrice** [roots/sage#2649]
* [x] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
* [x] This request isn't a duplicate of an existing issue
* [x] I've read the [docs](https://roots.io/documentation/) and followed them (if applicable)
* [x] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

## Description
With a fresh installation of the actual sage 10 with “composer create-project roots/sage your-theme-name dev-master”, and without changing anything (so with tailwind 2), compiling time with yarn start takes 6-9 seconds for a simple change in app.scss like body {background-color:red} (or with @apply...) and styles are not injected.

## Steps to reproduce

1. install sage 10 with “composer create-project roots/sage your-theme-name dev-master”
2. change your local url
3. make a css change in app.scss

**Expected behavior:** compiling time with yarn start should be much faster and styles change should be injected

**Actual behavior:** compiling time takes between 6 and 9 seconds for a very simple change and styles are not injected. I have to manually reload to see the change.

**Reproduces how often:** 100%

## Versions
mac os latest - on macbook pro 2018 with 16gb RAM
node 14.15.3
yarn 1.22.4
sage 10
flywheel local dev environment: 5-10.1 (latest)

## Additional information
Here are more info on several tentatives to solve this https://discourse.roots.io/t/sage-10-browsersync-issue-with-reload/20314
In summary: using postcss instead of sass is interesting but does not solve the problem. 

Any help is welcome. Thank you.

**Comment by knowler** [comment]
If you are causing Tailwind to compile a lot (using `@apply` a lot and/or writing custom CSS) it’s going to be slow. I recommend just using the utility classes, since then it never needs to compile again. 

The other alternative is to import your Sass or CSS via JS and then tell Mix to extract it to a file, but you have to write a bit of a custom webpack config for that. (This method is faster because if you break up your CSS then it shouldn’t compile the whole thing when you just change one file.)

I’m going to close this. If you’d like to discuss further, I recommend starting a topic on Roots Discourse.

**Comment by SergiArias** [comment]
I am interested too in the topic. I use mainly utility classes but from time to time I create a component or change some things in the config file and yes, it is quite slow.

In case you open a discussion for the specifics, please link it!

**Comment by Log1x** [comment]
> I am interested too in the topic. I use mainly utility classes but from time to time I create a component or change some things in the config file and yes, it is quite slow.
> 
> In case you open a discussion for the specifics, please link it!

This is a Tailwind+Webpack issue unfortunately. Tailwind has a new compiler in the works that should fix these issues dramatically.

**Comment by SergiArias** [comment]
yeah, they have [this new experimental library](https://blog.tailwindcss.com/just-in-time-the-next-generation-of-tailwind-css). I've seen the video and it's quite amazing, as you can build custom classes on the go.
I hope it gets implemented in Sage soon!!
